cmd/instance_show.go

```diff
@@ -93,106 +106,139 @@ func (c *instanceShowCmd) cmdRun(cmd *cobra.Command, _ []string) error {
+		CreationDate:       instance.CreatedAT.String(),
+		DiskSize:           humanize.IBytes(uint64(instance.DiskSize << 30)),
 		ElasticIPs:         make([]string, 0),
-		ID:                 *instance.ID,
-		IPAddress:          utils.DefaultIP(instance.PublicIPAddress, "-"),
-		IPv6Address:        utils.DefaultIP(instance.IPv6Address, "-"),
+		ID:                 instance.ID,
+		IPAddress:          utils.DefaultIP(&instance.PublicIP, "-"),
+		IPv6Address:        utils.DefaultIP(ipV6, "-"),
 		Labels: func() (v map[string]string) {
 			if instance.Labels != nil {
-				v = *instance.Labels
+				v = instance.Labels
 			}
 			return
 		}(),
-		Name:            *instance.Name,
+		Name:            instance.Name,
 		PrivateNetworks: make([]string, 0),
-		SSHKey:          utils.DefaultString(instance.SSHKey, "-"),
+		SSHKey:          utils.DefaultString(sshKeyName, "-"),
 		SecurityGroups:  make([]string, 0),
-		State:           *instance.State,
+		SecureBoot:      *instance.SecurebootEnabled,
+		Tpm:             *instance.TpmEnabled,
+		State:           instance.State,
 		Zone:            c.Zone,
 	}
 
 	out.PrivateInstance = "No"
-	if instance.PublicIPAssignment != nil && *instance.PublicIPAssignment == "none" {
+	if instance.PublicIPAssignment != "" && instance.PublicIPAssignment == "none" {
 		out.PrivateInstance = "Yes"
 	}
 
-	if instance.AntiAffinityGroupIDs != nil {
-		for _, id := range *instance.AntiAffinityGroupIDs {
-			antiAffinityGroup, err := globalstate.EgoscaleClient.GetAntiAffinityGroup(ctx, c.Zone, id)
+	if instance.AntiAffinityGroups != nil {
+		for _, group := range instance.AntiAffinityGroups {
+			resp, err := client.ListAntiAffinityGroups(ctx)
+			if err != nil {
+				return err
+			}
+			found_group, err := resp.FindAntiAffinityGroup(group.ID.String())
 			if err != nil {
 				return fmt.Errorf("error retrieving Anti-Affinity Group: %w", err)
 			}
-			out.AntiAffinityGroups = append(out.AntiAffinityGroups, *antiAffinityGroup.Name)
+			out.AntiAffinityGroups = append(out.AntiAffinityGroups, found_group.Name)
 		}
 	}
 
 	out.DeployTarget = "-"
-	if instance.DeployTargetID != nil {
-		DeployTarget, err := globalstate.EgoscaleClient.GetDeployTarget(ctx, c.Zone, *instance.DeployTargetID)
+	if instance.DeployTarget != nil {
+		resp, err := client.ListDeployTargets(ctx)
+		if err != nil {
+			return err
+		}
+		dt, err := resp.FindDeployTarget(instance.DeployTarget.ID.String())
 		if err != nil {
 			return fmt.Errorf("error retrieving Deploy Target: %w", err)
 		}
-		out.DeployTarget = *DeployTarget.Name
+		out.DeployTarget = dt.Name
 	}
 
-	if instance.ElasticIPIDs != nil {
-		for _, id := range *instance.ElasticIPIDs {
-			elasticIP, err := globalstate.EgoscaleClient.GetElasticIP(ctx, c.Zone, id)
+	if instance.ElasticIPS != nil {
+		for _, eip := range instance.ElasticIPS {
+			resp, err := client.ListElasticIPS(ctx)
+			if err != nil {
+				return err
+			}
+			found_eip, err := resp.FindElasticIP(eip.ID.String())
```

Comment:
```suggestion
			foundEip, err := resp.FindElasticIP(eip.ID.String())
```

[golang naming convention is to use camelCase or PascalCase (depending on context) for naming variables](https://go.dev/doc/effective_go#mixed-caps)